src/main/java/com/minecolonies/coremod/blocks/BlockPaperwall.java

```diff
@@ -183,9 +174,12 @@ protected BlockStateContainer createBlockState()
     }
 
     @Override
-    protected ItemStack getSilkTouchDrop(final IBlockState state)
+    public boolean canPaneConnectTo(final IBlockAccess world, final BlockPos pos, final EnumFacing dir)
     {
-        return new ItemStack(Item.getItemFromBlock(this), 1, state.getValue(VARIANT).getMetadata());
+        final BlockPos off = pos.offset(dir);
+        final IBlockState state = world.getBlockState(off);
+        return canPaneConnectToBlock(state.getBlock())
+                 || state.isSideSolid(world, off, dir.getOpposite()) || state.getBlock() instanceof BlockPaperwall;
     }
 
     public enum EnumType implements IStringSerializable
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 61 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2972)